### PR TITLE
rust raids update

### PR DESCRIPTION
**Pull request by MSilva95**

### CODE REVIEW

File: subjects/road_intersection/README.md [hunk 1]
```diff
@@ -32,15 +32,15 @@ _________________________|_______________________
...
+               |    ↓    | l  s  r |
                |    ↓    |    ↑    |
                |    ↓    |    ↑    |
                |    ↓    |    ↑    |
                |    ↓    |    ↑    |
                |    ↓    |    ↑    |
 ```
 
-For clarification reasons we will assume that a lane can have three different routes (consider you are in the vehicle position):
+For clarification reasons we will assume that a lane can have three different routes (consider that you are in the vehicle position):
```
Comment (davhojt) [inline subjects/road_intersection/README.md:43]: Does this mean that each direction (e.g. North->South) has three different routes?

The usage of the word 'lane' seems to contradict what is written on line 16:

https://github.com/01-edu/public/blob/332e922aaec357b2a8b5e9ecc25c8f929de95701/subjects/road_intersection/README.md?plain=1#L16

File: subjects/road_intersection/README.md [hunk 2]
```diff
@@ -65,15 +65,17 @@ You are free to decide what algorithm you want to implement for the traffic ligh
 
 Vehicles must obey this rules:
 
-- Vehicles must have a color depending on their route the colors are up to you to decide(ex:`r`- purple, `s`- Blue and `l`- Yellow). This must then be given during the audit
+- Vehicles must have a color depending on their route the colors are up to you to decide 
 
-- Autonomous, vehicles driving on a lane with a **given route** must follow the direction of
-  that route, its not possible for the driver to change lanes or route.
+- Vehicles must have a color so that it's possible to identify which route it will follow (ex:`r`- purple, `s`- Blue and `l`- Yellow). This information about the colors must be given to the auditor of the raid.
```
Comment (davhojt) [inline subjects/road_intersection/README.md:70]: Probably can merge line 68 and line 70.

https://github.com/01-edu/public/blob/332e922aaec357b2a8b5e9ecc25c8f929de95701/subjects/road_intersection/README.md?plain=1#L68
https://github.com/01-edu/public/blob/332e922aaec357b2a8b5e9ecc25c8f929de95701/subjects/road_intersection/README.md?plain=1#L70

File: subjects/road_intersection/README.md [hunk 3]
```diff
@@ -65,15 +65,17 @@ You are free to decide what algorithm you want to implement for the traffic ligh
...
+
+- Autonomous vehicles driving on a lane with a **given route** must follow the direction of
+that route, it's not possible for the driver to change lanes or routes.
```
Comment (davhojt) [inline subjects/road_intersection/README.md:73]: Autonomous vehicles don't have drivers.
Probable we can say something like:
> it is not possible for the autonomous vehicle to change lanes or routes.

File: subjects/road_intersection/README.md [hunk 4]
```diff
@@ -101,17 +103,20 @@ For this it must be possible to do the following:
...
 
-> Arrow keys must not let the user spam the creation of vehicles, vehicles must be created with a safe distance between them.
+> Arrow keys must not let the user spam the creation of vehicles, they must be created with a safe distance between them.
+
+> A safe distance is any distance that allows the vehicles not to crash into each other.
 
 ### Example
 
-You can see an example [here](https://www.youtube.com/watch?v=6B0-ZBET6mo).
+You can see an example for the road_intersection [here](https://www.youtube.com/watch?v=6B0-ZBET6mo).
```
Comment (davhojt) [inline subjects/road_intersection/README.md:112]: This shows an example which does not feature a four  lane intersection.
Should we clarify that this only shows traffic lights & collision avoidance?

File: subjects/road_intersection/audit/README.md [hunk 5]
```diff
@@ -1,86 +1,98 @@
...
 ###### Was a vehicle generated from the cardinal east?
 
 ##### Try pressing the `"R"` key.
 
-###### Are vehicles generated randomly(random lane and route)?
+###### Was the vehicle generated randomly (random lane and route)?
+
+##### Try pressing the `"R"` key more than 5 times to generate multiple vehicles in different lines.
+
+###### Can you confirm that the vehicles were created and maintain a safe distance one from another?
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:29]: > one from another?

should be
> from one another?

File: subjects/road_intersection/audit/README.md [hunk 6]
```diff
@@ -1,86 +1,98 @@
...
+###### Did all the vehicles passed the intersection without any collision?
 
-##### Try to generate five vehicles using the `"Up"` key, at the same time generate two vehicles using the key `"Right"`.
+##### Try to generate five vehicles using the `"Up"` key, at the same time generate two vehicles using the `"Right"` key.
 
-###### Did all vehicles passed the intersection without any collision?
+###### Did all the vehicles passed the intersection without any collision?
 
-##### Try to generate one vehicle for all the lanes (to this at least 3 times).
+##### Try to generate one vehicle for all the lanes (do this at least 3 times).
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:63]: What is the context of "lanes" here?
If each direction has two lanes, can it be possible to certainly know which lane will be used?
For example, perhaps a vehicle can go straight from either lane.

File: subjects/road_intersection/audit/README.md [hunk 7]
```diff
@@ -1,86 +1,98 @@
...
 
-###### Did all vehicles passed the intersection without any collision?
+###### Did all the vehicles passed the intersection without any collision?
 
 ##### Try to generate vehicles randomly using the `"R"` key. Then wait for at least 1 min.
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:67]: This seems to be a less-specific duplicate of another question. It does have a 1 minute wait though, what does that test?

https://github.com/01-edu/public/blob/332e922aaec357b2a8b5e9ecc25c8f929de95701/subjects/road_intersection/audit/README.md?plain=1#L27

File: subjects/road_intersection/audit/README.md [hunk 8]
```diff
@@ -1,86 +1,98 @@
...
 
-###### Did all vehicles passed the intersection without any collision?
+###### Did all the vehicles passed the intersection without any collision?
 
-###### Was there a low traffic congestion while running the application? (ex: high traffic congestion can be 8 or more vehicles in the same lane without proceeding)?
+##### Try to generate multiple vehicles with lanes at your choice.
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:71]: Same as the comment for line 63 (although I may have misunderstood the concept of lanes).

File: subjects/road_intersection/audit/README.md [hunk 9]
```diff
@@ -1,86 +1,98 @@
...
 
-###### Are vehicles generated randomly(random lane and route)?
+###### Was the vehicle generated randomly (random lane and route)?
+
+##### Try pressing the `"R"` key more than 5 times to generate multiple vehicles in different lines.
+
+###### Can you confirm that the vehicles were created and maintain a safe distance one from another?
 
-##### Try to generate tree vehicles in the same lane (do this to all lanes).
+##### Try to generate tree vehicles in the same lane (do this on all lanes one at a time).
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:31]: > Try to generate tree vehicles...

Should this be:
> Try to generate three vehicles...

Comment (davhojt) [inline subjects/road_intersection/audit/README.md:31]: Same comment as the one for line 63.

File: subjects/road_intersection/audit/README.md [hunk 10]
```diff
@@ -1,86 +1,98 @@
...
 
-###### Do vehicles proceed when ever there is a green light?
+###### Do vehicles proceed whenever there is a green light?
 
 #### Bonus
 
-###### +Is there any type of image sprite for traffic light?
+###### +Is there any type of image sprite for traffic light? (An image sprite is a collection of images put into a single image.)
 
 ###### +Did the student implement some kind of animation and image sprite for the vehicle?
```
Comment (davhojt) [inline subjects/road_intersection/audit/README.md:98]: We don't seem to check if the project implements a 4-lane intersection.

**Review by davhojt — COMMENTED** [review]
I left a few comments about the context of the usage of 'lanes'.
It could be that I have misunderstood if the project is supposed to implement a 4-lane intersection or not.

File: subjects/road_intersection/README.md (hunk 4 above)
Comment (MSilva95) [reply subjects/road_intersection/README.md:112]: it has for lanes, the example shows 2 roads and four lanes

File: subjects/road_intersection/audit/README.md (hunk 7 above)
Comment (MSilva95) [reply subjects/road_intersection/audit/README.md:67]: one is to confirm if the R feature is working and the cars are created at a safe distance, and the other is to check if there is any crash when generating vehicles with the R

File: subjects/road_intersection/audit/README.md (hunk 10 above)
Comment (MSilva95) [reply subjects/road_intersection/audit/README.md:98]: There is no need for it as we ask from line 7 to 19 if the prog creates a vehicle in each of the lanes

File: subjects/road_intersection/audit/README.md (hunk 6 above)
Comment (MSilva95) [reply subjects/road_intersection/audit/README.md:63]: lanes == traffic lanes
You can say that each road has two lanes in this program

File: subjects/road_intersection/audit/README.md (hunk 5 above)
Comment (MSilva95) [reply subjects/road_intersection/audit/README.md:29]: done

File: subjects/road_intersection/audit/README.md (hunk 9 above)
Comment (MSilva95) [reply subjects/road_intersection/audit/README.md:31]: done

File: subjects/road_intersection/README.md (hunk 3 above)
Comment (MSilva95) [reply subjects/road_intersection/README.md:73]: done

File: subjects/road_intersection/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/road_intersection/README.md:70]: done

File: subjects/road_intersection/README.md [hunk 11]
```diff
@@ -49,11 +49,11 @@ For clarification reasons we will assume that a lane can have three different ro
 2. Traffic lights
 
 Traffic lights are signalize devices positioned at road intersections that follows an universal color code,
-normally its green, red and amber, but for this project you will just use the colors **red** and **green**.
+it's usually green, red and amber, but for this project you will just use the colors **red** and **green**.
 
 You will then have to create some kind of representation for the traffic lights and distribute them for each lane in the intersection.
 
-You are free to decide what algorithm you want to implement for the traffic light system, but keep in mind that traffic congestion should not be to high.
+You are free to decide what algorithm you want to implement to represent traffic light system, but keep in mind that traffic congestion should not be to high (8 or more vehicles).
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/road_intersection/README.md:56]: done

File: subjects/road_intersection/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/road_intersection/README.md:43]: Yes, each lane has the possibility to go three different ways.
i specified line 16 better

File: subjects/drawing/README.md [hunk 12]
```diff
@@ -2,44 +2,50 @@
...
 
-#### Bonus
+You are free to implement all the shapes with whatever internal structure you see fit, but you must provide an associated function `new` for all the shapes, which will be described below:
 
-- Implement drawing a pentagon (implement the structure Pentagon, and the trait needed to draw in the image)
+- `Point`: a new point should be created from two `i32` values.
+- `Line`: a new line should be created from references to two different points. Also it will define an associated function called `random`:
+    - `random`: receives two arguments, the first is the `maximum x` value a point can have and the second is the `maximum y` value that a point can have.
+- `Triangle`: a new triangle should be created from references to three different points.
+- `Rectangle`: a new rectangle should be created from two references to different points. ??????????????????????????????
```
[the remark this answers is not in the document]
Comment (MSilva95) [reply subjects/drawing/README.md:37]: i thought i had removed them sorry

File: subjects/road_intersection/README.md (hunk 1 above)
Comment (davhojt) [reply subjects/road_intersection/README.md:45]: We're using 2-lane highways. My comment is irrelevant.

File: subjects/road_intersection/audit/README.md (hunk 10 above)
Comment (davhojt) [reply subjects/road_intersection/audit/README.md:98]: My comment is irrelevant. We're using single lane carriageways.

File: subjects/road_intersection/audit/README.md (hunk 9 above)
Comment (davhojt) [reply subjects/road_intersection/audit/README.md]: My comment is irrelevant. We're using single carriageways.

File: subjects/road_intersection/audit/README.md (hunk 8 above)
Comment (davhojt) [reply subjects/road_intersection/audit/README.md]: My comment is irrelevant. We're using single carriageways.

File: subjects/road_intersection/audit/README.md (hunk 6 above)
Comment (davhojt) [reply subjects/road_intersection/audit/README.md:63]: My comment is irrelevant. We're using single carriageways.

File: subjects/road_intersection/README.md (hunk 4 above)
Comment (davhojt) [reply subjects/road_intersection/README.md:111]: My comment is irrelevant. We're using single carriageways.

**Pull request merged**